examples/twittermap/web/public/javascripts/searchbar/controllers.js

```diff
@@ -4,7 +4,64 @@ angular.module('cloudberry.util', ['cloudberry.common'])
       if ($scope.keyword && $scope.keyword.trim().length > 0) {
         cloudberry.parameters.keywords = $scope.keyword.trim().split(/\s+/);
         // skip the empty query for now.
-        cloudberry.query(cloudberry.parameters);
+
+          //stopword filtering feature
+          String.prototype.removeStopWords = function() {
+              var x;
+              var y;
+              var word;
+              var stop_word;
+              var regex_str;
+              var regex;
+              var cleansed_string = this.valueOf();
+              var stop_words = external_stop_words();
+
+              // Split out all the individual words in the phrase
+              var words = cleansed_string.match(/[^\s]+|\s+[^\s+]$/g);
+
+              // Review all the words
+              for(x=0; x < words.length; x++) {
+                  // For each word, check all the stop words
+                  for(y=0; y < stop_words.length; y++) {
+                      // Get the current word
+                      word = words[x].replace(/\s+|[^a-z]+/ig, "");   // Trim the word and remove non-alpha
```

Comment:
Why we need to remove non-alpha numeric characters? if someone asks "can123", do we want to check "can"? I think we need to have a discussion with @chenlica.